app/javascript/components/Home.jsx

```diff
@@ -1,73 +1,96 @@
-import React, { useEffect, useState } from "react";
+import React, { useState } from 'react'
 
-export default () => {
+import Error from './shared/Error'
+import Table from './companies/Table'
+import Pagination from './shared/Pagination'
+import FilterForm from './companies/FilterForm'
+
+import { useDebouncedEffect } from '../utils/hooks'
+
+const Home = () => {
   // List of fetched companies
-  const [companies, setCompanies] = useState([]);
+  const [companies, setCompanies] = useState([])
+
+  // Initial page load flag
+  const [isInitialMount, setIsInitialMount] = useState(true)
+
+  // Errors
+  const [error, setError] = useState(null)
+
+  // Pagination
+  const [currentPage, setCurrentPage] = useState(1)
+  const [totalPages, setTotalPages] = useState(1)
 
   // Table filters
-  const [companyName, setCompanyName] = useState("");
-  const [industry, setIndustry] = useState("");
-  const [minEmployee, setMinEmployee] = useState("");
-  const [minimumDealAmount, setMinimumDealAmount] = useState("");
+  const [companyName, setCompanyName] = useState('')
+  const [industry, setIndustry] = useState('')
+  const [minEmployee, setMinEmployee] = useState('')
+  const [minimumDealAmount, setMinimumDealAmount] = useState('')
+
+  const handlePageChange = (page) => {
+    setCurrentPage(page)
+    fetchCompanies(page)
+  }
+
+  const queryParams = new URLSearchParams({ companyName, industry, minEmployee, minimumDealAmount }).toString()
+  const companiesUrl = `/api/v1/companies?${queryParams}`
```

Comment:
I feel that here are too many state variables and we could improve that further but for the time being I left them like that